node_modules/oae-content/lib/api.js

```diff
@@ -1079,48 +1079,75 @@ var setContentPermissions = module.exports.setContentPermissions = function(ctx,
             return callback(err);
         }
 
-        // Check if the current user can manage the content
-        ContentAuthz.canManageContent(ctx, contentObj, function(err, canManage) {
+        // Check that the user isn't trying to assign any role higher than his own
+        AuthzAPI.resolveEffectiveRole(ctx, contentObj.id, contentObj.tenant.alias, contentObj.visibility, ContentConstants.roles.ALL_PRIORITY, function(err, effectiveRole, canInteract) {
             if (err) {
                 return callback(err);
-            } else if (!canManage) {
-                return callback({'code': 401, 'msg': 'You are not allowed to manage this piece of content'});
+            } else if (!canInteract) {
+                return callback({'code': 401, 'msg': 'You are not allowed to interact with this piece of content'});
             }
 
-            // Ensure that we're not crossing tenant boundaries or violating principal visibility settings
-            _checkNewContentPermissions(ctx, contentObj, permissionChanges, function(err, membershipAfterChanges, newMemberIds, updatedMemberIds, removedMemberIds, newMembers) {
+            ContentAuthz.canManageContent(ctx, contentObj, function(err, canManage) {
                 if (err) {
                     return callback(err);
                 }
 
-                // Apply the permission changes to the content. This will then update the content libraries as well
-                _setContentPermissions(ctx, contentObj, permissionChanges, function(err, newContentObj) {
-                    if (err) {
-                        return callback(err);
+                AuthzAPI.getDirectRoles(_.keys(permissionChanges), contentObj.id, function(err, existingRoles) {
+                    if (!canManage) {
+                        if (_.contains(permissionChanges, false)) {
+                            return callback({'code': 401, 'msg': 'Only managers can remove access'});
+                        }
+
+                        var settingHigherRole = _.find(permissionChanges, function(role, principalId) {
+                            return ContentConstants.roles.ALL_PRIORITY.indexOf(effectiveRole) < ContentConstants.roles.ALL_PRIORITY.indexOf(role);
+                        });
+                        if (settingHigherRole) {
+                            return callback({'code': 401, 'msg': 'You cannot set another user\'s role higher than your own'});
+                        }
+                        var hasHigherRole = _.find(existingRoles, function(role, principalId) {
+                            return ContentConstants.roles.ALL_PRIORITY.indexOf(effectiveRole) < ContentConstants.roles.ALL_PRIORITY.indexOf(role);
+                        });
+                        if (hasHigherRole) {
+                            return callback({'code': 401, 'msg': 'You cannot change the role of someone with a higher role than your own'});
+                        }
                     }
```

Comment:
I think we can have a `ContentAuthz.canSetContentPermissions` that takes care of doing the `resolveEffectiveRole` check and the role promotion checks to ensure roles aren't being set higher than one's own role. A few notes about this process:

* The documentation of the method should be clear that it's only checking permissions surrounding access to the content item and the role promotions, and it's *not* checking permissions against the users being promoted (e.g., it won't catch if the user is trying to add a loggedin user from another tenant). That stuff is done in `checkNewContentPermissions`. Ideally it *would* do this, but it's probably too big of a bite to take right now.
* We don't need to do the `ContentAuthz.canManageContent` check. Instead, we can look at the effective role that came out of `resolveEffectiveRole` and see if `effectiveRole === ContentConstants.roles.MANAGER`